### PR TITLE
Sessions

### PR DESCRIPTION
**Pull request by Steve-D-Eckles**
-Update some file / variable names to match standard convention
-Update route structure for teacher pages and added a teacher home page
-Give course overview its own endpoint
-Add a session view and several views related to the creation and editing of course sessions
-Add tests for all new code
-Add test for overlooked function in db.py

### CODE REVIEW

File: portal/auth.py [hunk 1]
```diff
@@ -36,8 +36,7 @@ def index():
             session['user_role'] = user['role']
             #Return different views for a teacher vs student
             if user['role'] == 'teacher':
-                return "Hello teacher"
-                #return redirect(url_for('portal.teacher_page'))
+                return redirect(url_for('teacher.home'))
             elif user['role'] == 'student':
                 # return "hello student"
```
Comment (kevinlueke) [inline portal/auth.py:41]: Needs to be deleted

File: portal/templates/create-sessions.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <form action="{{ url_for('teacher.session_submit') }}" method="post">
+    <!-- Set dates and times for the session -->
+    <label for="meeting_days">Meeting Days:</label>
+    <input type="text" name="meeting_days" maxlength="6" size="8">
+    <!-- Set a name for the session -->
+    <label for="session_name">Session Code</label>
+    <input type="text" name="session_name" maxlength="1" size="2">
+    <!-- Create the session -->
+    <button type="submit">Create Session</button>
+    <button formaction="{{ url_for('teacher.session_cancel') }}" type="submit" formmethod="get" name="button">Cancel Session</button>
```
Comment (kevinlueke) [inline portal/templates/create-sessions.html:42]: Not to get picky, but a button should always have a value as well as a name.

File: portal/templates/edit-sessions.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {% endif %}
+  </form>
+
+  <form action="{{ url_for('teacher.session_submit') }}" method="post">
+    <!-- Set dates and times for the session -->
+    <label for="meeting_days">Meeting Days:</label>
+    <input type="text" name="meeting_days" maxlength="6" size="8" value="{{ session_info['meeting_days'] }}">
+    <!-- Set a name for the session -->
+    <label for="session_name">Session Code</label>
+    <input type="text" name="session_name" maxlength="1" size="2" value="{{ session_info['session_name']}}">
```
Comment (kevinlueke) [inline portal/templates/edit-sessions.html:39]: Line 39, towards the end, after "session_name']" needs a space.

File: portal/templates/edit-sessions.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <!-- Create the session -->
+    <button type="submit">Confirm Edits</button>
+    <button formaction="{{ url_for('teacher.session_cancel') }}" type="submit" formmethod="get" name="button">Cancel Edit</button>
```
Comment (kevinlueke) [inline portal/templates/edit-sessions.html:42]: Name value pair needed

**Review by kainguyen123-oss — COMMENTED** [review]
The code is solid good use of Joins, good edits to the HTML, good comments and good tests.

**Review by tjusticelee — COMMENTED** [review]
Looks like a lot of thought and effort was put into this code. From what I can comprehend it looks good.

File: tests/data.sql [hunk 5]
```diff
@@ -3,3 +3,22 @@
...
 VALUES ('teacher@stevenscollege.edu', 'pbkdf2:sha256:150000$YKTR53od$3956727084a9d3470d800ca00005a2258bde5affab208eb9ae211c46a28f575d', 'teacher', 'Fedjona', 'Zachiberto', 'CSET'),
        ('student@stevenscollege.edu', 'pbkdf2:sha256:150000$db49oioE$95b37dfa834211006b45fddfa03debdca5d2ccf49486bb7e35af144406d51c8a', 'student', 'Lueklee', 'Kevstice', 'CSET');
+
+INSERT INTO courses (course_code, course_name, major, description, teacher_id)
+VALUES ('180', 'Big Software Energy', 'CSET', 'A class that really makes you FEEL like a developer', 1);
+
+INSERT INTO sessions (session_name, meeting_days, course_id)
+VALUES ('A', 'MTWThF', 1);
+
+-- ID,EMAIL,PASSWORD,NAME,ROLE,MAJOR
```
Comment (kevinlueke) [inline tests/data.sql:13]: Probably don't want to have user information in here. Especially passwords... Instead we could just include this info in a file on our Trello board so it is accessible if needed.

**Review by kainguyen123-oss — APPROVED** [review]

**Pull request merged**